uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -226,6 +226,20 @@ define([
       };
 
       /**
+       * Checks if user can see file list in files tab. It needs to check for
+       * soft delete flag on leave request as well for files.
+       *
+       * @return {Boolean} true is user can see file list
+       */
+      this.canShowFileList = function () {
```

Comment:
Why can't you just show, in the UI, just the files that do not have `toBeDeleted` set to `true`?, something like (pseudo code)

```html
<ul class="chr_leave-request-files_list container-fluid" ng-if="$ctrl.canShowFileList()">
  <li ng-repeat="file in $ctrl.request.files | filter: { toBeDeleted: false }">
```
This way the list is simply either empty (so it looks like it's not there) or with some elements, but we don't rely on some additional method on the component